zmq/src/ZMQ.mli

```diff
@@ -1,290 +1,10 @@
-  val set_plain_username : 'a t -> string -> unit
-  val get_plain_username : 'a t -> string
-  val set_plain_password : 'a t -> string -> unit
-  val get_plain_password : 'a t -> string
-  val set_curve_server : 'a t -> bool -> unit
-  val set_curve_publickey : 'a t -> string -> unit
-  val get_curve_publickey : 'a t -> string
-  val set_curve_secretkey : 'a t -> string -> unit
-  val get_curve_secretkey : 'a t -> string
-  val set_curve_serverkey : 'a t -> string -> unit
-  val get_curve_serverkey : 'a t -> string
-  val get_mechanism : 'a t -> [`Null | `Plain | `Curve]
-  val set_zap_domain : 'a t -> string -> unit
-  val get_zap_domain : 'a t -> string
-  val set_conflate : [> `Pull | `Push | `Sub | `Pub | `Dealer] t -> bool -> unit
-
-  val get_fd : 'a t -> Unix.file_descr
-
-  type event = No_event | Poll_in | Poll_out | Poll_in_out | Poll_error
-  val events : 'a t -> event
-
-end
-
-module Proxy : sig
-  val create: ?capture:[> `Pub|`Dealer|`Push|`Pair] Socket.t -> 'a Socket.t -> 'b Socket.t -> unit
-end
-
-module Poll : sig
-
-  type t
-
-  type poll_event = In | Out | In_out
-  type 'a poll_mask = ('a Socket.t * poll_event)
-
-  val mask_of : 'a poll_mask array -> t
-  val poll : ?timeout: int -> t -> poll_event option array
-
-end
-
-module Monitor : sig
-  type t
-
-  type address = string
-  type error_no = int
-  type error_text = string
-
-  type event =
-  | Connected of address * Unix.file_descr
-  | Connect_delayed of address
-  | Connect_retried of address * int (*interval*)
-  | Listening of address * Unix.file_descr
-  | Bind_failed of address * error_no * error_text
-  | Accepted of address * Unix.file_descr
-  | Accept_failed of address * error_no * error_text
-  | Closed of address * Unix.file_descr
-  | Close_failed of address * error_no * error_text
-  | Disconnected of address * Unix.file_descr
-
-  val create: 'a Socket.t -> t
-  val connect: Context.t -> t -> [>`Monitor] Socket.t
-
-  (** Receive an event from the monitor socket.
-      block indicates if the call should be blocking or non-blocking. Default true
-  *)
-  val recv: ?block:bool -> [> `Monitor ] Socket.t -> event
-
-  val string_of_event: event -> string
-
-  (** Create a memorizing function for converting an event to a string.
-      As its it not possible to reliably retrieve the peer address of a closed socket
-      dues to a race condition, this function pairs connects and disconnects and returns the matching
-      connect peer address to disconnects.
-
-      Note that it is not possible to retrieve the peer address of connect events is the peer has disconnected
-      before string_of_event is called
-  *)
-  val mk_string_of_event: unit -> (event -> string)
-
-end
-
-module Z85 : sig
-  val encode : string -> string
-  val decode : string -> string
-end
-
-module Curve : sig
-  (** [keypair ()] returns a pair [public, secret] of Z85 encoded keys. *)
-  val keypair : unit -> string * string
+include module type of struct
```

Comment:
Very interesting, I didn't know about this. I think we should for now allow mixed use since two libraries might depend on `zmq`, one having already upgraded to the new name, the other not yet.

When we remove the deprecated `ZMQ` module this can be then rolled back to stop allowing it, since it will not be possible to generate `ZMQ.*` types anyway and it will make the code less head-scratchy. I think putting a comment in this place would be a reasonable thing to do.